server/controllers/conversation.js

```diff
@@ -0,0 +1,142 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+const Conversation = require('../models/Conversation');
+const Message = require('../models/Message');
+const User = require('../models/User');
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { from, to } = req.body;
+
+  try {
+    if (!from || !to) {
+      res.status(400).json({
+        error: 'from and to are required',
+      });
+      throw new Error('from and to are required');
+    }
+
+    const fromUser = await User.findOne({ email: from });
```

Comment:
This step can be avoided as we will be receiving the sender and receiver's user id's from the request body

Reply:
Corrected throughout